### DISCUSSION TITLE
Identify test end

### DISCUSSION THREAD

**Discussion by ACBNair** [jtpereyda/boofuzz#602] [Q&A] [answered]
I see that session.Fuzz is a blocking call and it won't give control (after test finish) unless user press any key. Is there any events to know that the test ended? I would like to end some monitors right when the test finished. Currently they continue monitoring till user provide keyboard interrupt.

**Comment by cq674350529** [comment]
After looking at the code, there might a way to achieve your purpose. In the `_main_fuzz_loop()`, if the fuzzing session completed, the `session.end_time` will be assigned. So you may check this variable if it's `None` in your monitor.

https://github.com/jtpereyda/boofuzz/blob/dbe720a425d8ade45accad317d8e4f0c66912ae3/boofuzz/sessions.py#L1402-L1412

**Comment by SR4ven** [comment, marked as answer]
See https://github.com/jtpereyda/boofuzz/discussions/600#discussioncomment-2262236
You can either disable the webinterface completely or simply make session.fuzz return immediately after testing ended with `keep_web_open = False`.
